src/main/cc/maze/cell/cell.h

```diff
@@ -2,8 +2,29 @@
 #define CC_CELL_CELL_H_
 
 // Abstract class to represent a cell.
-public class Cell {
-	 	
+class Cell {
+	// Creates a Cell object at (x,y)
+	Cell(int x, int y);
+
+	// Gets the x offsest of a Cell
+	int GetLocationX();
+
+	// Gets the y offset of a Cell
+	int GetLocationY();
+
+	// Gets the status of whether a Cell has been visited
+	bool IsVisited();
+
+	// Sets a Cell to visited (true)
+	void VisitCell();
+
+	// Sets a Cell to unvisited (false) 
+	void unVisitCell();
+
+private:
+	int x;
```

Comment:
I might give these different names like x_loc_ or x_loc or something of the sort so they wont get confused with local variables used inside functions